library/src/main/kotlin/com/chuckerteam/chucker/internal/ui/MainActivity.kt

```diff
@@ -55,18 +75,54 @@ internal class MainActivity :
             tutorialLink.movementMethod = LinkMovementMethod.getInstance()
             transactionsRecyclerView.apply {
                 setHasFixedSize(true)
-                addItemDecoration(DividerItemDecoration(this@MainActivity, DividerItemDecoration.VERTICAL))
+                addItemDecoration(
+                    DividerItemDecoration(
+                        this@MainActivity,
+                        DividerItemDecoration.VERTICAL
+                    )
+                )
                 adapter = transactionsAdapter
             }
         }
 
         viewModel.transactions.observe(
-            this,
-            { transactionTuples ->
-                transactionsAdapter.submitList(transactionTuples)
-                mainBinding.tutorialGroup.isVisible = transactionTuples.isEmpty()
+            this
+        ) { transactionTuples ->
+            transactionsAdapter.submitList(transactionTuples)
+            mainBinding.tutorialGroup.isVisible = transactionTuples.isEmpty()
+        }
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            handleNotificationsPermission()
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.TIRAMISU)
+    private fun handleNotificationsPermission() {
+        when {
+            ContextCompat.checkSelfPermission(
+                this, Manifest.permission.POST_NOTIFICATIONS
+            ) == PackageManager.PERMISSION_GRANTED -> {
+                Logger.info("Notification permission granted")
+            }
+            shouldShowRequestPermissionRationale(Manifest.permission.POST_NOTIFICATIONS) -> {
+                Snackbar.make(
+                    mainBinding.root,
+                    applicationContext.getString(R.string.chucker_notifications_permission_not_granted),
+                    Snackbar.LENGTH_LONG
+                ).setAction(applicationContext.getString(R.string.chucker_change)) {
+                    Intent(Settings.ACTION_APPLICATION_DETAILS_SETTINGS).apply {
+                        addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+                        data = Uri.fromParts("package", packageName, null)
+                    }.also { intent ->
+                        startActivity(intent)
+                    }
+                }.show()
```

Comment:
Here we're breaking a bit from the Google's guideline where the rationale should just explain why we need the permission and we should still allow the user to pick yes/no. If yes -> we should fire the `permissionRequest`.

Letting the user jump to the Settings screen sort of assumes that the user selected No on the past.

Reply:
Yes, I did it on purpose as I assume that Chucker is used by devs, not regular users, so showing one more dialog to somebody, who disabled permission on purpose on first launch is going to be annoying. Devs know how and why permission is needed, especially, since showing notifications is what Chucker does. 
Instead, you have a snackbar and can change settings if you wish or just mind your own business with Chucker without dismissing the dialog.